pkg/sink/sink.go

```diff
@@ -136,3 +167,96 @@ func createRequestURI(apiVersion, namePlural, namespace string) string {
 	uri = path.Join(uri, namePlural)
 	return uri
 }
+
+func (r Resource) validateEvent(triggerValidate *triggersv1.TriggerValidate, headers http.Header, payload []byte) error {
+	tr, err := r.createValidateTask(triggerValidate, headers, payload)
+	if err != nil {
+		return nil
+	}
+
+	tr, err = r.PipelineClient.TektonV1alpha1().TaskRuns(r.EventListenerNamespace).Create(tr)
+	if err != nil {
+		return err
+	}
+
+	for i := taskRunPollingRetries; i >= 0; i-- {
+		tr, err := r.PipelineClient.TektonV1alpha1().TaskRuns(r.EventListenerNamespace).Get(tr.Name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		if tr.IsSuccessful() {
+			break
+		}
+
+		if tr.IsDone() && !tr.IsSuccessful() {
+			return errors.New("validation taskrun: " + tr.Name + " failed")
+		}
+
+		time.Sleep(taskRunPollingInterval)
+	}
+	return nil
+}
+
+func (r Resource) createValidateTask(triggerValidate *triggersv1.TriggerValidate,
+	headers http.Header, payload []byte) (*pipelinev1.TaskRun, error) {
+	// Checking whether task define in taskref exists or not
+	task, err := r.PipelineClient.TektonV1alpha1().Tasks(r.EventListenerNamespace).Get(triggerValidate.TaskRef.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	if task.Spec.Inputs == nil {
+		return nil, errors.New("parameters are mandatory for validate task")
+	}
+
+	params := []pipelinev1.Param{}
+	params = append(params, triggerValidate.Params...)
+	params = append(params, pipelinev1.Param{
+		Name: "Payload",
+		Value: pipelinev1.ArrayOrString{
+			Type:      pipelinev1.ParamTypeString,
+			StringVal: string(payload),
+		},
+	})
+
+	for _, v := range task.Spec.Inputs.Params {
+		h, ok := headers[v.Name]
+		if !ok {
```

Comment:
I think we should check to see if there were default values within the task. Otherwise, this parameter will be never be passed and fail.